src/translators/ObjectsTranslator.ts

```diff
@@ -185,42 +188,51 @@ export abstract class ObjectsTranslator {
             for (let i = 0; i < numTableModifications; i++) {
                 const objectDefinition = []; // object definition will store one or more modification objects
 
-                const originalId = outBufferToJSON.readChars(4),
-                    customId = outBufferToJSON.readChars(4),
-                    modificationCount = outBufferToJSON.readInt();
-
-                for (let j = 0; j < modificationCount; j++) {
-                    const modification: Modification = {
-                        id: '',
-                        type: ModificationType.string,
-                        level: 0,
-                        column: 0,
-                        value: {}
-                    };
-
-                    modification.id = outBufferToJSON.readChars(4);
-                    modification.type = this.varTypes[outBufferToJSON.readInt()]; // 'int' | 'real' | 'unreal' | 'string',
+                const originalId = outBufferToJSON.readChars(4);
+                const customId = outBufferToJSON.readChars(4);
+                let sets = 1;
+                const setsFlag: number[] = [];
+                if (fileVersion >= 3) {
+                    sets = outBufferToJSON.readInt();
+                }
 
-                    if (type === ObjectType.Doodads || type === ObjectType.Abilities || type === ObjectType.Upgrades) {
-                        modification.level = outBufferToJSON.readInt();
-                        modification.column = outBufferToJSON.readInt();
+                for (let set = 0; set < sets; set++) {
+                    if(fileVersion >=3) {
+                        setsFlag[set] = outBufferToJSON.readInt();
```

Comment:
Is `setsFlag` information we want to store in the JSON object to write it back to a w3-file later?

It doesn't look like you're using this variable anywhere.

Reply:
Could probably remove as we don't patch existing object Giles, only generate new ones